src/send/SelectRecipientButtons.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from 'react'
+import { useAsync } from 'react-async-hook'
+import { useTranslation } from 'react-i18next'
+import { Platform } from 'react-native'
+import { PERMISSIONS, PermissionStatus, RESULTS, check, request } from 'react-native-permissions'
+import { SendEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { phoneNumberVerifiedSelector } from 'src/app/selectors'
+import Dialog from 'src/components/Dialog'
+import SelectRecipientButton from 'src/components/SelectRecipientButton'
+import QRCode from 'src/icons/QRCode'
+import Social from 'src/icons/Social'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import useSelector from 'src/redux/useSelector'
+import Logger from 'src/utils/Logger'
+import { navigateToPhoneSettings } from 'src/utils/linking'
+
+const CONTACTS_PERMISSION =
+  Platform.OS === 'ios' ? PERMISSIONS.IOS.CONTACTS : PERMISSIONS.ANDROID.READ_CONTACTS
+
+type Props = {
+  onContactsPermissionGranted: () => void
+}
+
+export default function SelectRecipientButtons({ onContactsPermissionGranted }: Props) {
+  const { t } = useTranslation()
+  const phoneNumberVerified = useSelector(phoneNumberVerifiedSelector)
+  const [contactsPermissionStatus, setContactsPermissionStatus] = useState<
+    PermissionStatus | undefined
+  >(undefined)
+  const [showConnectPhoneNumberModal, setShowConnectPhoneNumberModal] = useState(false)
+  const [showEnableContactsModal, setShowEnableContactsModal] = useState(false)
+  const [navigateToPhoneVerification, setNavigateToPhoneVerification] = useState(false)
+
+  useAsync(async () => {
+    if (!contactsPermissionStatus) {
+      setContactsPermissionStatus(await check(CONTACTS_PERMISSION))
+    }
+  }, [contactsPermissionStatus])
```

Comment:
why would you want this? looks like `setContactsPermissionStatus` is always setting it to the output of `await check(CONTACTS_PERMISSION)`, which if you believe the typing for that function, should never be undefined

Reply:
so isn't it enough just to set it once?

Reply:
I thought we needed to specify all reactive values used in the hook have to be specified similar to useEffect, but the linter doesn't seem to complain, so maybe my understanding was wrong.

Reply:
we should specify all dependent variables in the dependency array so that stale values are not used. there's a linter rule that we can enable for this, but it would complain about empty dependency arrays 😅 we have genuine use cases for an empty array though so we haven't enabled this linting rule, but perhaps we should...